wcfsetup/install/files/lib/acp/form/LanguageAddForm.class.php

```diff
@@ -103,149 +75,123 @@ public function readParameters()
+            ...$this->locales,
+        ];
+
+        $this->form->appendChildren([
+            FormContainer::create('data')
+                ->appendChildren([
+                    TextFormField::create('languageName')
+                        ->label('wcf.global.name')
+                        ->description('wcf.acp.language.name.description')
+                        ->maximumLength(255)
+                        ->required(),
+                    TextFormField::create('languageCode')
+                        ->label('wcf.acp.language.code')
+                        ->description('wcf.acp.language.code.description')
+                        ->maximumLength(20)
+                        ->required()
+                        ->addValidator(new FormFieldValidator('unique', function (TextFormField $formField) {
+                            if ($formField->getValidationErrors() !== []) {
+                                return;
+                            }
+
+                            if (
+                                $this->formObject instanceof Language
+                                && \mb_strtolower($this->formObject->languageCode) === $formField->getValue()
+                            ) {
+                                return;
+                            }
+
+                            if (LanguageFactory::getInstance()->getLanguageByCode($formField->getValue())) {
+                                $formField->addValidationError(new FormFieldValidationError(
+                                    'notUnique',
+                                    'wcf.acp.language.add.languageCode.error.notUnique'
+                                ));
+                            }
+                        })),
+                    TextFormField::create('countryCode')
+                        ->label('wcf.acp.language.countryCode')
+                        ->description('wcf.acp.language.countryCode.description')
+                        ->maximumLength(10)
+                        ->required(),
+                    SingleSelectionFormField::create('locale')
+                        ->label('wcf.acp.language.locale')
+                        ->description('wcf.acp.language.locale.description')
+                        ->options($locales)
+                        ->addValidator(new FormFieldValidator(
+                            'locale',
+                            static function (SingleSelectionFormField $formField) {
+                                if ($formField->getValidationErrors() !== []) {
+                                    return;
+                                }
+
+                                $languageCodeField = $formField->getDocument()->getNodeById('languageCode');
+                                \assert($languageCodeField instanceof TextFormField);
+                                if ($languageCodeField->getValue() === '') {
+                                    return;
+                                }
```

Comment:
```suggestion
                                if ($formField->getValue() === '') {
                                    return;
                                }

                                $languageCodeField = $formField->getDocument()->getNodeById('languageCode');
                                \assert($languageCodeField instanceof TextFormField);
```

You didn't use the suggestion as-is: Currently the empty selection is rejected, but it should not (as it's not required).